Runtime/Highlight/ObjectHighlighter.cs

```diff
@@ -0,0 +1,453 @@
+﻿using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Innoactive.Creator.Unity;
+
+namespace Innoactive.Creator.BasicInteraction
+{
+    /// <summary>
+    /// Adds highlighting functionality to a GameObject with Renderers.
+    /// </summary>
+    /// <remarks>
+    /// Highlights are always queued following a LIFO (Last In First Out) scheme. 
+    /// </remarks>
+    public class ObjectHighlighter : MonoBehaviour
+    {
+        private readonly Dictionary<string, Material> highlightIDs = new Dictionary<string, Material>();
```

Comment:
after 3-4 minutes into the class i still have no clue what this does. Might there be a better name for it which explains is better?